packages/asset-swapper/src/utils/market_operation_utils/path_optimizer.ts

```diff
@@ -275,16 +275,14 @@ function findRoutesAndCreateOptimalPath(
         }
 
         // TODO(kimpers): remove once we have solved the rounding/precision loss issues in the Rust router
-        const scaleOutput = (fillInput: BigNumber, output: BigNumber) =>
-            output
-                .dividedBy(fillInput)
-                .times(rustInputAdjusted)
-                .decimalPlaces(0, side === MarketOperation.Sell ? BigNumber.ROUND_FLOOR : BigNumber.ROUND_CEIL);
+        const maxSampledOutput = BigNumber.max(...routeSamples.map(s => s.output));
+        const scaleOutput = (output: BigNumber) => BigNumber.min(output.times(scale), maxSampledOutput);
```

Comment:
Wouldn't we want to min against the sum of outputs? We can still use multiple sources/samples even if we don't hit our Target, right?

Reply:
This is all a mess and if it weren't for the f64 precision issues we wouldn't need this duct tape.

I could be wrong but this scaling is on a per fill level, and therefore it should always be from a single source, within the sampled range of that source, correct?

I added this min to check that we don't scale the output outside of the sampled range, where the liquidity is undefined.